x/observer/keeper/blame.go

```diff
@@ -40,6 +41,21 @@ func (k Keeper) GetAllBlame(ctx sdk.Context) (BlameRecords []*types.Blame, found
 	return
 }
 
+func (k Keeper) GetBlameByChainAndNonce(ctx sdk.Context, chainID int64, nonce int64) (BlameRecords []*types.Blame, found bool) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefix(types.BlameKey))
+	blamePrefix := fmt.Sprintf("%d-%d", chainID, nonce)
```

Comment:
I looked into the code, I see that ZetaClient index the blame index with:
```
func digestReq(request keygen.Request) (string, error) {
	bytes, err := json.Marshal(request)
	if err != nil {
		return "", err
	}

	hasher := sha3.NewLegacyKeccak256()
	hasher.Write(bytes)
	digest := hex.EncodeToString(hasher.Sum(nil))

	return digest, nil
}
```
Therefore, I'm not sure to understand how they would be indexed with `[chainID]-[nonce]`

Reply:
This particular digest is for keygen. There won't be any chain or nonce associated with it, I can however add a different prefix to this like "keygen-" to differentiate it. 

The change was really intended for keysign failures.